### DISCUSSION TITLE
newline-before-return rule for .jsx components

### DISCUSSION THREAD

**Discussion by 1alexvash** [eslint/eslint#16732] [Q&A]
Is there a similar rule to [this one](https://eslint.org/docs/latest/rules/newline-before-return)? But for JSX components

I want to enforce a blank return line for every JSX component like so:

Example of incorrect code ❌ 
```typescript
const Component = () => {
  // Some statefulll logic...
  return <div/>;
};
```

Example of correct code ✔️ 
```typescript
const Component = () => {
  // Some statefulll logic...

  return <div/>;
};
```

**Comment by 1alexvash** [comment]
I mean, state is a logic part, and return is a UI part

So for code style convention, they should be decoupled

**Comment by sam3k** [comment]
@1alexvash I took a quick look at [eslint-plugin-react](https://github.com/jsx-eslint/eslint-plugin-react) and it does not seem like they have* a similar rule. Since it is a plugin that is not maintained by Eslint; maybe try raising an [issue](https://github.com/jsx-eslint/eslint-plugin-react/issues) there?

**Reply by 1alexvash** [reply]
Um, sure

Should I just create a copy of this ticket, and post it there?

**Reply by Rec0iL99** [reply]
Hi @1alexvash, like Sam mentioned you can open a new issue in the [eslint-plugin-react](https://github.com/jsx-eslint/eslint-plugin-react) repo explaining the rule clearly that you would like to have implemented. You can give the same details that you gave here.

**Reply by 1alexvash** [reply]
Okay, let's do it!
